storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3904,10 +3910,16 @@ class Rdb_transaction {
   virtual void release_snapshot(TABLE_TYPE table_type) = 0;
 
   bool has_snapshot(TABLE_TYPE table_type) const {
+    assert(m_read_opts[INTRINSIC_TMP].snapshot == nullptr);
```

Comment:
Hey Laurynas, I wonder why the snapshot for intrinsic tmp table is always null?

Reply:
Temp tables are always private to the owning connection and don't need to participate in isolation concerns